src/Runner/Runner.php

```diff
@@ -124,9 +124,10 @@ public function fix()
             $this->cacheManager
         );
 
-        $collection = $this->linter->isAsync()
-            ? new FileCachingLintingIterator($fileFilteredFileIterator, $this->linter)
-            : new FileLintingIterator($fileFilteredFileIterator, $this->linter);
+        $collection = new FileLintingIterator(
+            new \CachingIterator($fileFilteredFileIterator),
```

Comment:
could you please describe what is a benefit of using `CachingIterator` in code you proposed ?

Reply:
The CachingIterator was used before (just via extended and even somewhat re-implemented which I considered superfluous so I removed that one and replaced it with this one).

Reply:
so, you didn't answer my question.
what is the benefit of using `CachingIterator` in code you are proposing?

I will answer you - none.

That "superfluous" class you have removed was responsible for running async linter on 2nd core, allowing one core lint file A and other core to fix file B.
With your changes, you drop the parallelism. Sometimes, it's better to understand before dropping the code.

Reply:
@keradus: Core as processor cores? I was indeed not aware of that. Caching both do actually help to have the same behaviour. I'll update the PR.

Reply:
yes, cores as cpu cores.
I think you don't understand what `Caching` in `CachingIterator` stands for.
Using CachingIterator for non-async linter will only increase memory usage, without giving any benefit